packages/headless/src/controllers/commerce/search-box/headless-search.box.ts

```diff
@@ -0,0 +1,114 @@
+import {fetchQuerySuggestions} from '../../../features/query-suggest/query-suggest-actions';
+import {logSearchboxSubmit} from '../../../features/query/query-analytics-actions';
+import {executeSearch} from '../../../features/search/search-actions';
+import {SuggestionHighlightingOptions, Delimiters} from '../../../utils/highlight';
+import {Controller} from '../../controller/headless-controller';
+import {
+  buildCoreSearchBox,
+  SearchBoxState,
+  Suggestion,
+} from '../../core/search-box/headless-core-search-box';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+
+export type {
+  SearchBoxState,
+  SuggestionHighlightingOptions,
+  Suggestion,
+  Delimiters,
+};
+
+export interface SearchBoxOptions {
+  /**
+   * A unique identifier for the controller.
+   * By default, a unique random identifier is generated.
+   */
+  id?: string;
+
+  /**
+   * Specifies delimiters to highlight parts of a query suggestion that e.g match, do not match the query.
+   */
+  highlightOptions?: SuggestionHighlightingOptions;
+
+  // TODO(nico): Figure out what to do with this
```

Comment:
I believe we could just add a case in the commerceFacetSetReducer (facet-set-slice.ts) to deselect all facet values when the `deselectAllBreadcrumbs` action in dispatched and that would do the trick for now...

Reply:
Handled this here: https://github.com/coveo/ui-kit/pull/3461/files#diff-41be90ca28bd75d552f2ea907b277dd78bd8e11b0501469a78eb2fab064869d1R101-R104